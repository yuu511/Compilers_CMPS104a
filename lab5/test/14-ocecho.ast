ROOT "<<ROOT>>" 0.0.0 {0} 
FUNCTION "" 6.22.1 {0} void function 
TYPE_ID "" 6.22.1 {0} 
VOID "void" 6.22.1 {0} 
IDENT "fail" 6.22.6 {0} 
PARAM "(" 6.22.11 {0} 
TYPE_ID "" 6.22.12 {1} string variable param lval 
STRING "string" 6.22.12 {0} 
IDENT "expr" 6.22.19 {0} 
TYPE_ID "" 6.22.25 {1} string variable param lval 
STRING "string" 6.22.25 {0} 
IDENT "file" 6.22.32 {0} 
TYPE_ID "" 6.22.38 {1} int variable param lval 
INT "int" 6.22.38 {0} 
IDENT "line" 6.22.42 {0} 
FUNCTION "" 6.24.1 {1} void function 
TYPE_ID "" 6.24.1 {0} 
VOID "void" 6.24.1 {0} 
IDENT "putchr" 6.24.6 {0} 
PARAM "(" 6.24.13 {0} 
TYPE_ID "" 6.24.14 {2} int variable param lval 
INT "int" 6.24.14 {0} 
IDENT "chr" 6.24.18 {0} 
FUNCTION "" 6.25.1 {2} void function 
TYPE_ID "" 6.25.1 {0} 
VOID "void" 6.25.1 {0} 
IDENT "putint" 6.25.6 {0} 
PARAM "(" 6.25.13 {0} 
TYPE_ID "" 6.25.14 {3} int variable param lval 
INT "int" 6.25.14 {0} 
IDENT "num" 6.25.18 {0} 
FUNCTION "" 6.26.1 {3} void function 
TYPE_ID "" 6.26.1 {0} 
VOID "void" 6.26.1 {0} 
IDENT "putstr" 6.26.6 {0} 
PARAM "(" 6.26.13 {0} 
TYPE_ID "" 6.26.14 {4} string variable param lval 
STRING "string" 6.26.14 {0} 
IDENT "str" 6.26.21 {0} 
FUNCTION "" 6.28.1 {4} int function 
TYPE_ID "" 6.28.1 {0} 
INT "int" 6.28.1 {0} 
IDENT "getchr" 6.28.5 {0} 
PARAM "(" 6.28.11 {0} 
FUNCTION "" 6.29.1 {5} string function 
TYPE_ID "" 6.29.1 {0} 
STRING "string" 6.29.1 {0} 
IDENT "getstr" 6.29.8 {0} 
PARAM "(" 6.29.14 {0} 
FUNCTION "" 6.30.1 {6} string function 
TYPE_ID "" 6.30.1 {0} 
STRING "string" 6.30.1 {0} 
IDENT "getln" 6.30.8 {0} 
PARAM "(" 6.30.13 {0} 
FUNCTION "" 7.5.1 {7} int function 
TYPE_ID "" 7.5.1 {0} 
INT "int" 7.5.1 {0} 
IDENT "main" 7.5.5 {0} 
PARAM "(" 7.5.9 {0} 
TYPE_ID "" 7.5.10 {8} int variable param lval 
INT "int" 7.5.10 {0} 
IDENT "argc" 7.5.14 {0} 
TYPE_ID "" 7.5.20 {8} string array variable param lval 
ARRAY "array" 7.5.20 {0} 
STRING "string" 7.5.26 {0} 
IDENT "argv" 7.5.34 {0} 
BLOCK "{" 7.5.40 {8} 
TYPE_ID "" 7.6.4 {8} int variable local lval vreg (7.6.4)
INT "int" 7.6.4 {0} 
IDENT "argi" 7.6.8 {0} 
INTCON "1" 7.6.15 {8} int variable local lval vreg 
WHILE "while" 7.7.4 {8} 
LT "<" 7.7.16 {8} int vreg 
IDENT "argi" 7.7.11 {8} int variable local lval (7.6.4)
IDENT "argc" 7.7.18 {8} int variable param lval (7.5.10)
BLOCK "{" 7.7.24 {8} 
IF "if" 7.8.7 {8} 
GT ">" 7.8.16 {8} int vreg 
IDENT "argi" 7.8.11 {8} int variable local lval (7.6.4)
INTCON "1" 7.8.18 {8} int const 
CALL "(" 7.8.28 {1} void function (6.24.1)
IDENT "putchr" 7.8.21 {0} 
CHARCON "' '" 7.8.29 {8} int const 
CALL "(" 7.9.14 {3} void function (6.26.1)
IDENT "putstr" 7.9.7 {0} 
INDEX "[" 7.9.19 {8} string lval vaddr 
IDENT "argv" 7.9.15 {8} string array variable param lval (7.5.20)
IDENT "argi" 7.9.20 {8} int variable local lval (7.6.4)
'=' "=" 7.10.12 {8} int variable local lval vreg 
IDENT "argi" 7.10.7 {8} int variable local lval (7.6.4)
'+' "+" 7.10.19 {8} int vreg 
IDENT "argi" 7.10.14 {8} int variable local lval (7.6.4)
INTCON "1" 7.10.21 {8} int const 
CALL "(" 7.12.11 {3} void function (6.26.1)
IDENT "putstr" 7.12.4 {0} 
STRINGCON ""\n"" 7.12.12 {8} string const 
RETURN "return" 7.13.4 {8} 
INTCON "0" 7.13.11 {8} int const 
