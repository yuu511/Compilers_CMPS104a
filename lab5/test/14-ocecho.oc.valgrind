==24216== Memcheck, a memory error detector
==24216== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==24216== Using Valgrind-3.14.0 and LibVEX; rerun with -h for copyright info
==24216== Command: ../oc 14-ocecho.oc
==24216== 
==24216== 
==24216== HEAP SUMMARY:
==24216==     in use at exit: 0 bytes in 0 blocks
==24216==   total heap usage: 488 allocs, 488 frees, 145,649 bytes allocated
==24216== 
==24216== All heap blocks were freed -- no leaks are possible
==24216== 
==24216== For counts of detected and suppressed errors, rerun with: -v
==24216== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
