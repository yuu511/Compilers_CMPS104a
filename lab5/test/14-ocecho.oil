main:      .function int 
           .param int argc
           .param ptr argv
           .local int argi
           argi = 1  
.wh0:      goto .od0 if argi > argc
.do0:
.if0:      goto .fi0 if argi < 1
.th0:      call putchr(' ')  
.fi0:
           $t0:c = argv[argi * :c]  
           call putstr($t0:c)  
           $t1:i = argi + 1
           argi = $t1:i  
           goto .wh0  
.od0
           call putstr("\n")  
           return 0
           return
           .end
