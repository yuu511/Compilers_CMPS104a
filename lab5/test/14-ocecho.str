string_set[   0]:    7296690267450279874 0x55fc62503cd8->"chr"
                   13348923631108115385 0x55fc62504698->"putstr"
string_set[   5]:   12679825168387489523 0x55fc625041f8->"num"
string_set[   6]:    9317282556992397945 0x55fc625062f8->"if"
string_set[   7]:    2297668033614959926 0x55fc62507278->"0"
                    6814028903552918164 0x55fc62505818->"array"
string_set[   8]:   18172887160629263681 0x55fc62506758->"' '"
string_set[  10]:    1583576014053361350 0x55fc62504858->"str"
                   13497578857207247412 0x55fc62503278->"string"
                   11597697714117577063 0x55fc62505d78->"="
string_set[  11]:    1396094175042499165 0x55fc625058d8->"<"
string_set[  13]:   12868445110721718657 0x55fc625038f8->")"
string_set[  16]:   11911796782241914524 0x55fc62501fe8->"void"
                   10159970873491820195 0x55fc62505e38->"1"
string_set[  18]:   13581035385482130682 0x55fc625068d8->"["
string_set[  19]:    5344662657899890615 0x55fc625039b8->";"
string_set[  20]:   15816506932648033945 0x55fc625031b8->"("
                   11868120863286193964 0x55fc62505478->"main"
string_set[  23]:    5266873771696973564 0x55fc62505cb8->"argi"
string_set[  25]:   11166701219533395393 0x55fc62504e78->"getstr"
string_set[  29]:   14635384904416252381 0x55fc625071b8->"return"
                    1370730858159036685 0x55fc62506eb8->"+"
string_set[  34]:    7211130878751977441 0x55fc62503b18->"putchr"
string_set[  37]:   12291392672316406109 0x55fc62503818->"line"
string_set[  39]:   18413295556186398326 0x55fc62504038->"putint"
string_set[  40]:   10319632053171904911 0x55fc62505f98->"while"
string_set[  44]:      21606623943142546 0x55fc62505b78->"{"
string_set[  45]:    6142509188972423790 0x55fc625033f8->""
string_set[  47]:   13686721134187162920 0x55fc62507068->"}"
string_set[  51]:    3701972582333163920 0x55fc62505638->"argc"
string_set[  53]:   12051031859636400188 0x55fc62506a18->"]"
string_set[  57]:    9386170080689500517 0x55fc62503338->"expr"
string_set[  58]:    4017852018407011111 0x55fc625035d8->"file"
                    3729804957429652673 0x55fc624fccc8->"<<ROOT>>"
string_set[  59]:   16579916199511856337 0x55fc62505178->"getln"
string_set[  61]:   14643508047410943861 0x55fc62505a78->"argv"
string_set[  63]:    9611428142580219387 0x55fc625030b8->"fail"
string_set[  64]:   12597418120160561823 0x55fc62503758->"int"
string_set[  70]:    9201800968376702500 0x55fc62504b78->"getchr"
string_set[  71]:   14526079891305879776 0x55fc62505a18->">"
string_set[  72]:   17138325012746839705 0x55fc625070a8->""\n""
string_set[  78]:    4706206694200543916 0x55fc625030f8->","
load_factor = 0.532
bucket_count = 79
max_bucket_size = 3
