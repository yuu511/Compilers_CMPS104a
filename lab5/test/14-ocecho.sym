main (7.5.1) {7} int function 
   argc (7.5.10) {8} int variable param lval 0
   argv (7.5.20) {8} string array variable param lval 1
   argi (7.6.4) {8} int variable local lval 2

getstr (6.29.1) {5} string function 

putchr (6.24.1) {1} void function 

getchr (6.28.1) {4} int function 

putstr (6.26.1) {3} void function 

getln (6.30.1) {6} string function 

fail (6.22.1) {0} void function 

putint (6.25.1) {2} void function 

