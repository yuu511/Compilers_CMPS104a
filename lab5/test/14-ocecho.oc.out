CONDITIONAL WHILE STATEMENT SYMBOL TOK_LT 
CONDITIONAL IF STATEMENT SYMBOL TOK_GT 
